Libs/MRML/DisplayableManager/vtkMRMLModelDisplayableManager.cxx

```diff
@@ -2041,36 +2052,49 @@ ::CreateTransformedClipper(vtkMRMLTransformNode *tnode, vtkMRMLModelNode::MeshTy
     vtkMatrix4x4::Multiply4x4(transformToWorld.GetPointer(), sliceMatrix, mat.GetPointer());
     planeDirection = (this->Internal->YellowSliceClipState == vtkMRMLClipModelsNode::ClipNegativeSpace) ? -1 : 1;
     this->SetClipPlaneFromMatrix(mat.GetPointer(), planeDirection, yellowSlicePlane.GetPointer());
-
-    if (type == vtkMRMLModelNode::UnstructuredGridMeshType)
+    }
+  else
+    {
+    slicePlanes = this->Internal->SlicePlanes;
+    }
+  if (type == vtkMRMLModelNode::UnstructuredGridMeshType)
+    {
+    if (this->Internal->ClippingMethod == vtkMRMLClipModelsNode::Straight)
       {
-      vtkExtractGeometry* clipper = vtkExtractGeometry::New();
-      //clipper->SetValue(0.0);
-      clipper->SetImplicitFunction(slicePlanes.GetPointer());
+      vtkClipDataSet* clipper = vtkClipDataSet::New();
+      clipper->SetClipFunction(slicePlanes);
       return clipper;
       }
     else
       {
-      vtkClipPolyData* clipper = vtkClipPolyData::New();
-      clipper->SetValue(0.0);
-      clipper->SetClipFunction(slicePlanes.GetPointer());
+      vtkExtractGeometry* clipper = vtkExtractGeometry::New();
+      clipper->SetImplicitFunction(slicePlanes);
+		  clipper->ExtractInsideOff();
```

Comment:
FYI: there were tabs here instead of spaces. I corrected it in the svn commit so np. Thank you.